controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -1264,6 +1260,180 @@ def test_controller_get_capabilities(self):
+        self.request.capacity_range.required_bytes = 11
+
+        self.servicer.ControllerExpandVolume(self.request, self.context)
+
+        self.assertEqual(self.context.code, grpc.StatusCode.OUT_OF_RANGE)
+        self.mediator.expand_volume.assert_not_called()
+
+    def _test_no_expand_needed(self):
+        response = self.servicer.ControllerExpandVolume(self.request, self.context)
+
+        self.assertEqual(self.context.code, grpc.StatusCode.OK)
+        self.assertFalse(response.node_expansion_required)
+        self.assertEqual(response.capacity_bytes, self.volume_before_expand.capacity_bytes)
+        self.mediator.expand_volume.assert_not_called()
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_size_zero(self, storage_agent):
+        self._prepare_expand_volume_mocks(storage_agent)
+        self.request.capacity_range.required_bytes = 0
+        self._test_no_expand_needed()
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_size_already_in_range(self, storage_agent):
+        self._prepare_expand_volume_mocks(storage_agent)
+        self.request.capacity_range.required_bytes = 2
+        self._test_no_expand_needed()
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_succeeds(self, storage_agent):
+        self._prepare_expand_volume_mocks(storage_agent)
+
+        response = self.servicer.ControllerExpandVolume(self.request, self.context)
+
+        self.assertEqual(self.context.code, grpc.StatusCode.OK)
+        self.assertTrue(response.node_expansion_required)
+        self.assertEqual(response.capacity_bytes, self.volume_after_expand.capacity_bytes)
+        self.mediator.expand_volume.assert_called_once_with(volume_id=self.volume_id,
+                                                            required_bytes=self.capacity_bytes)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_bad_id(self, storage_agent):
+        self._prepare_expand_volume_mocks(storage_agent)
+        self.request.volume_id = "123"
+
+        self.servicer.ControllerExpandVolume(self.request, self.context)
+
+        self.assertEqual(self.context.code, grpc.StatusCode.INVALID_ARGUMENT)
+        self.mediator.expand_volume.assert_not_called()
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_first_object_not_found(self, storage_agent):
+        self._prepare_expand_volume_mocks(storage_agent)
+        self.mediator.get_object_by_id.side_effect = [None, None]
+
+        self.servicer.ControllerExpandVolume(self.request, self.context)
+
+        self.assertEqual(self.context.code, grpc.StatusCode.NOT_FOUND)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_second_object_not_found(self, storage_agent):

```

Comment:
I wouldn't consider the expanded volume a "second object".
perhaps `test_expand_volume_not_found_before_expansion` and `test_expand_volume_not_found_after_expansion`

Reply:
done